### DISCUSSION TITLE
Redaction Processor from v0.40.0

### DISCUSSION THREAD

**Discussion by lalitmathwani** [open-telemetry/opentelemetry-collector-contrib#6596] [Q&A] [answered]
Hello all,

I am trying to experiment with using the redaction processor for our Otel deployment here @ Adobe. We have seen the usage of a few sensitive tokens in span attributes that we wanted to mask before sending to Jaeger.
Looking at using the redaction processor from the latest image  of contrib (https://hub.docker.com/layers/otel/opentelemetry-collector-contrib/0.40.0/images/sha256-b28b6c4e64429e46d6a327f33a95fafa1647439041acd519c66b6b63cc104392?context=explore). 
Tried configuring the redaction processor:
redaction:
    # Flag to allow all span attribute keys. Setting this to true disables the
    # allowed_keys list. The list of blocked_values is applied regardless. If
    # you just want to block values, set this to true.
    allow_all_keys: true
    # Blocklist for span attribute values
    blocked_values:
      - regexes for our sensitive tokens ..
     
and including it in a trace pipeline :
service:
  extensions: [ pprof, zpages, health_check ]
  pipelines:
    traces:
      receivers: [ otlp ]
      processors: [ batch, redaction ]
      exporters: [ logging, zipkin, jaeger ]
    metrics:
      receivers: [ otlp ]
      processors: [ batch ]
      exporters: [ logging, prometheus ]

Keep running into the following error:
Error: cannot unmarshal the configuration: unknown processors type "redaction" for "redaction"
2021/12/07 02:21:27 collector server run finished with error: cannot unmarshal the configuration: unknown processors type "redaction" for "redaction"

From git history, the redaction processor does seem like a fairly recent addition (started 14 days back - just 2 days before the v0.40.0 was cut):
https://github.com/open-telemetry/opentelemetry-collector-contrib/commits/main/processor/redactionprocessor

Wanted to check if it's ready to be used from the 0.40.0 image , or potentially I am doing something silly with the config above?

Cheers!

**Comment by jpkrohling** [comment, marked as answer]
This one hasn't been finished yet: #6495.

**Reply by lalitmathwani** [reply]
Thanks a lot @jpkrohling ! We would definitely be very interested in leveraging it as soon as it's done and released 👍 .
Could we ask for a rough estimate on when it is expected to be completed ? Thanks again ...

**Reply by jpkrohling** [reply]
No promises, there's even a discussion on whether the functionality of this processor shouldn't be merged with another one (attributes processor).

**Reply by lalitmathwani** [reply]
Got it , thanks a lot @jpkrohling. Either way we will keep an eye out for this enhancement.

**Reply by Harnoor-se7en** [reply]
@lalitmathwani how are you redacting the values currently? and @jpkrohling sorry to ask you again but do we have any rough timeline estimation for this at this point in time?

**Comment by lalitmathwani** [comment]
@Harnoor-se7en , we decided to extend the otel instrumentation lib and obfuscate sensitive values before exporting to the collector.
We followed the approach suggested by @trask here (many thanks @trask!) - https://github.com/open-telemetry/opentelemetry-java-instrumentation/discussions/4832

**Reply by Harnoor-se7en** [reply]
Thanks, @lalitmathwani. This indeed is a great solution and nicely written code for custom implementations. Unfortunately, we don't use opentelemetry-java-instrumentation and we're currently exploring ways to redact at the OTEL level.
